### ISSUE TITLE
Incorporate margin cache in cross-match calculation

### ISSUE THREAD

**Issue by delucchi-cmu** [astronomy-commons/lsdb#51]
This likely needs a design doc.

Load margin caches like with a normal catalog. but when do we merge it into the main catalog for use in cross-matching? potentially need to duplicate operations on the margin cache that have been performed on the primary catalog.

**Comment by mjuric** [comment]
The demo we're aiming for is to be able to run something like:

```
gaia = lsdb.table("gaia")
ztf = lsdb.table("ztf", ztf_margin)
ztf.box(0, 0, 1, 1).crossmatch(gaia, r=1*arcsec)
```

and compare it to a brute-force computed version of the same cross-match, and show they're the same.

**Comment by smcguire-cmu** [comment]
Closed with #144
